### PR TITLE
APM updates badge in React hook

### PR DESCRIPTION
**Pull request by ogupte**
Uses a custom React hook for badge updates in APM

### CODE REVIEW

**Review by sorenlouv — APPROVED** [review]
Great improvement with hooks!

File: x-pack/plugins/apm/public/components/app/Main/useUpdateBadgeEffect.ts [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+            }),
+            tooltip: i18n.translate('xpack.apm.header.badge.readOnly.tooltip', {
+              defaultMessage: 'Unable to save'
+            }),
+            iconType: 'glasses'
+          }
+        : undefined
+    );
+  }, []);
+};
```
Comment (sorenlouv) [inline x-pack/plugins/apm/public/components/app/Main/useUpdateBadgeEffect.ts:29]: Is this component specific to APM? If not, perhaps it should be in `ui/capabilities`.

Comment (kobelb) [reply x-pack/plugins/apm/public/components/app/Main/useUpdateBadgeEffect.ts:29]: The read-only badges are currently application specific

**Comment by kobelb** [comment]
Thanks @ogupte !

**Pull request merged**